api/tests/Feature/DirectivesTest.php

```diff
@@ -128,4 +135,113 @@ public function testPluckFromObject(): void
             ],
         ]);
     }
+
+    // Ensure the canOnParent custom directive works as expected
+    public function testCanOnParent(): void
+    {
+        $admin = User::factory()->create([
+            'first_name' => 'Admin',
+            'last_name' => 'Test',
+            'email' => 'admin@test.com',
+            'sub' => 'admin@test.com',
+            'legacy_roles' => [ApiEnums::LEGACY_ROLE_ADMIN, ApiEnums::LEGACY_ROLE_APPLICANT]
+        ]);
+        $applicant = User::factory()->create([
+            'first_name' => 'Applicant',
+            'last_name' => 'Test',
+            'email' => 'applicant@test.com',
+            'sub' => 'applicant@test.com',
+            'legacy_roles' => [ApiEnums::LEGACY_ROLE_APPLICANT]
+        ]);
+        $otherApplicant = User::factory()->create([
+            'first_name' => 'Other',
+            'last_name' => 'Test',
+            'email' => 'other@test.com',
+            'sub' => 'other@test.com',
+            'legacy_roles' => [ApiEnums::LEGACY_ROLE_APPLICANT]
+        ]);
+
+        $candidates = PoolCandidate::factory()->count(3)->availableInSearch()->create([
+            'user_id' => $applicant->id,
+        ]);
+
+        $this->mockResolver(function () use ($applicant) {
+            return $applicant->fresh();
+        });
+
+        $this->schema = /** @lang GraphQL */ '
+        type PoolCandidate {
+            id: ID
+            user: User @belongsTo(relation: "user")
+        }
+        type User {
+            id: ID
+            poolCandidates: [PoolCandidate] @hasMany @canOnParent(ability: "view")
+        }
+        type Query {
+            user: User @mock
+        }
+        ';
+
+        $query = /** @lang GraphQL */
+        '
+        query getUser {
+            user {
+                id
+                poolCandidates {
+                    id
+                }
+            }
+        }
+        ';
+        $fullResponse = [
+            'data' => [
+                'user' => [
+                    'id' => $applicant->id,
+                    'poolCandidates' => $candidates->map(function ($candidate) {
+                        return ['id' => $candidate->id];
+                    })->toArray(),
+                ],
+            ],
+        ];
+        $unauthorizedResponse = [
+            'data' => [
+                'user' => [
+                    'id' => $applicant->id,
+                    'poolCandidates' => null,
+                ],
+            ],
+            'errors' => [
+                ['message' => 'This action is unauthorized.']
+            ]
+        ];
+
+        $isUser = function($user) {
+            return function($argument) use ($user) {
+                return $argument instanceof User && $argument->id == $user->id;
+            };
+        };
+        $mock = $this->mock(UserPolicy::class);
```

Comment:
Why is it necessary to mock the user policy?  Couldn't you use the real policy in the test schema?

Reply:
I originally let it use the real policy, but mocking it
1. makes this test independent of any changes to the UserPolicy, which is not what I'm trying to test here
2. and more importantly, allows me to confirm that UserPolicy was checked, not PoolCandidatePolicy, which is the key point of the directive.

Reply:
Makes sense, thanks. :pray: